File: DB/10월 4주차/타겟넘버.js
function solution(numbers, target) {
  let count = 0;

  function targetNumber(index, sum) {
    if (index === numbers.length) {
      if (sum === target) count++;
      return;
    }

    targetNumber(index + 1, sum + numbers[index]);
    targetNumber(index + 1, sum - numbers[index]);
  }

  targetNumber(0, 0);
  return count;
}

File: JY/10월 5주차/타겟넘버.java
import java.util.*;

class Solution {
    public static int answer= 0 ;
    public void bt(int level, int[] numbers, int target, int score){
        if(level == numbers.length){
            if(score == target)
                answer++;
            return;
        }
            bt(level+1, numbers, target, numbers[level] + score);
            bt(level+1, numbers, target,  score-  numbers[level]);
    }
    public int solution(int[] numbers, int target) {
        
        bt(0, numbers, target, 0);
        return answer;
    }
}